master_node:
  ros__parameters:
    # Whether to use cubic interpolation for offset
    use_interpolation: true
    # PID control parameters
    pid_pitch:
      kp: 10.0
      ki: 0.005
      integral_limit: 1.0
      integral_threshold: 1.0
    
    pid_yaw:
      kp: 10.0
      ki: 0.005
      integral_limit: 1.0
      integral_threshold: 1.0

    # Trigger servo motor push retract angles
    trigger:
      push: 166.0
      retract: 70.0
    
    cutoff_frequency: 5.0 # Hz
    
    # Motor control parameters
    stop_throttle: 0.1
    motor_offset: 0.003
    default_speed: 0.24 #4000 RPM
    
    # Launch threshold parameters
    N: 7
    EPS: 0.007
    max_consecutive_nans: 5
    
motor_node:
  ros__parameters:
    pitch_servo:
      cw_center_us: 1655.0 # DOWN
      ccw_center_us: 1905.0 # UP

    yaw_servo:
      cw_center_us: 1700.0 # LEFT
      ccw_center_us: 1830.0 # RIGHT

imu_node:
  ros__parameters:
    # Set true to show gyro x and gyro y values
    log_gyro: true

    gyro_offsets:
      x: 0.0
      y: 0.0
      z: 0.0